src/dune_engine/action_builder.ml

```diff
@@ -498,14 +498,22 @@ struct
         (f, deps)
       | Build b ->
         let* b, deps0 = exec b in
-        let+ r, deps1 = build_static_rule_deps_and_exec b in
+        let+ r, deps1 = build_all_static_deps_and_exec b in
```

Comment:
This looks like a change in semantics of the `Build` constructor. Is it intended? If yes, why and will users be able to see different behaviour?

Reply:
Got it, thanks.